src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,237 @@
+        this._boxEdgeB = createLine().classed("drag-edge-tb", true);
+        this._boxEdgeL = createLine().classed("drag-edge-lr", true);
+        this._boxEdgeR = createLine().classed("drag-edge-lr", true);
+
+        var createCorner = () => this._box.append("circle")
+                                     .attr("r", this._boxEdgeWidth / 2)
+                                     .style({
+                                       "opacity": 0,
+                                       "fill": "pink"
+                                     });
+        this._boxCornerTL = createCorner().classed("drag-corner-tl", true);
+        this._boxCornerTR = createCorner().classed("drag-corner-tr", true);
+        this._boxCornerBL = createCorner().classed("drag-corner-bl", true);
+        this._boxCornerBR = createCorner().classed("drag-corner-br", true);
+      }
+
+      private _getEdges(p: Point) {
+        var edges = {
+          top: false,
+          bottom: false,
+          left: false,
+          right: false
+        };
+
+        var bounds = this.bounds();
+        var t = bounds.topLeft.y;
+        var b = bounds.bottomRight.y;
+        var l = bounds.topLeft.x;
+        var r = bounds.bottomRight.x;
+        var he = this._boxEdgeWidth / 2;
+
+        if (l - he <= p.x && p.x <= r + he) {
+          edges.top = (t - he <= p.y && p.y <= t + he);
+          edges.bottom = (b - he <= p.y && p.y <= b + he);
+        }
+
+        if (t - he <= p.y && p.y <= b + he) {
+          edges.left = (l - he <= p.x && p.x <= l + he);
+          edges.right = (r - he <= p.x && p.x <= r + he);
+        }
+
+        return edges;
+      }
+
+      public _doRender() {
+        super._doRender();
+        if (this.boxVisible()) {
+          var bounds = this.bounds();
+          var t = bounds.topLeft.y;
+          var b = bounds.bottomRight.y;
+          var l = bounds.topLeft.x;
+          var r = bounds.bottomRight.x;
+
+          this._boxEdgeT.attr({
+            x1: l, y1: t, x2: r, y2: t
+          });
+          this._boxEdgeB.attr({
+            x1: l, y1: b, x2: r, y2: b
+          });
+          this._boxEdgeL.attr({
+            x1: l, y1: t, x2: l, y2: b
+          });
+          this._boxEdgeR.attr({
+            x1: r, y1: t, x2: r, y2: b
+          });
+
+          this._boxCornerTL.attr({ cx: l, cy: t });
+          this._boxCornerTR.attr({ cx: r, cy: t });
+          this._boxCornerBL.attr({ cx: l, cy: b });
+          this._boxCornerBR.attr({ cx: r, cy: b });
+        }
+      }
+
+      /**
+       * Gets the edge width of the drag box.
+       *
+       * @return {number} The edge width of the drag box.
+       */
+      public edgeWidth(): number;
+      /**
+       * Sets the edge width of the drag box.
+       *
+       * @param {number} width The desired edge width.
+       * @return {DragBoxLayer} The calling DragBoxLayer.
+       */
+      public edgeWidth(width: number): DragBoxLayer;
+      public edgeWidth(width?: number): any {
```

Comment:
Now that we are in interactive land, we can think about these names in terms of interactivity possibly.  edgeWidth to me still sounds like the edge with of the actual box (stroke).  What's a better name?